src/lsm/compaction.zig

```diff
@@ -250,22 +264,22 @@ pub fn CompactionType(
             assert(compaction.filter.state == .building);
             assert(compaction.data.state == .building);
 
-            // TODO Implement manifest.move_table() optimization if there's only range.table_count == 1.
-            // This would do update_tables + insert_tables inline without going through the iterators.
-
-            const iterator_b_context = .{
-                .grid = grid,
-                .manifest = manifest,
-                .level = level_b,
-                .snapshot = op_min,
-                .key_min = range.key_min,
-                .key_max = range.key_max,
-                .direction = .ascending,
-                .table_info_callback = iterator_b_table_info_callback,
-            };
-
-            compaction.iterator_a.start(iterator_a_context, iterator_a_io_callback);
-            compaction.iterator_b.start(iterator_b_context, iterator_b_io_callback);
+            // Start the compaction iterators if we need to merge tables between levels.
+            if (compaction.strategy == .table_merge) {
```

Comment:
This is potentially dangerous, if another enum is added, since we don't exhaustively switch.